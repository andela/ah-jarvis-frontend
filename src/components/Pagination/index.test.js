import { shallow } from 'enzyme';
import React from 'react';
import Paginator from '.';

describe('<Paginator />', () => {
  it('renders  <Paginator /> components', () => {
    const snap = shallow(<Paginator
      activePage={1}
      total={4}
      onClick={() => null}
    />);
    expect(snap).toMatchSnapshot();
  });
});
