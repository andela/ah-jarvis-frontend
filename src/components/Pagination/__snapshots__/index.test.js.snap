// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paginator /> renders  <Paginator /> components 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Paginator
    activePage={1}
    onClick={[Function]}
    total={4}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="pagination center m-b--20"
>
        <li
                className="disabled"
        >
                <a
                        href="#!"
                        onClick={[Function]}
                >
                        <i
                                className="material-icons"
                        >
                                chevron_left
                        </i>
                </a>
        </li>
        <li
                className="active"
        >
                <a
                        href="#!"
                        onClick={[Function]}
                >
                        1
                </a>
        </li>
        <li
                className="disabled"
        >
                <a
                        href="#!"
                        onClick={[Function]}
                >
                        <i
                                className="material-icons"
                        >
                                chevron_right
                        </i>
                </a>
        </li>
</ul>,
      "className": "row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="disabled"
>
            <a
                        href="#!"
                        onClick={[Function]}
            >
                        <i
                                    className="material-icons"
                        >
                                    chevron_left
                        </i>
            </a>
</li>,
          Array [
            <li
              className="active"
>
              <a
                            href="#!"
                            onClick={[Function]}
              >
                            1
              </a>
</li>,
          ],
          <li
            className="disabled"
>
            <a
                        href="#!"
                        onClick={[Function]}
            >
                        <i
                                    className="material-icons"
                        >
                                    chevron_right
                        </i>
            </a>
</li>,
        ],
        "className": "pagination center m-b--20",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#!"
              onClick={[Function]}
>
              <i
                            className="material-icons"
              >
                            chevron_left
              </i>
</a>,
            "className": "disabled",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="material-icons"
>
                chevron_left
</i>,
              "href": "#!",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "chevron_left",
                "className": "material-icons",
              },
              "ref": null,
              "rendered": "chevron_left",
              "type": "i",
            },
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#!"
              onClick={[Function]}
>
              1
</a>,
            "className": "active",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
              "href": "#!",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 1,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#!"
              onClick={[Function]}
>
              <i
                            className="material-icons"
              >
                            chevron_right
              </i>
</a>,
            "className": "disabled",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="material-icons"
>
                chevron_right
</i>,
              "href": "#!",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "chevron_right",
                "className": "material-icons",
              },
              "ref": null,
              "rendered": "chevron_right",
              "type": "i",
            },
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="pagination center m-b--20"
>
          <li
                    className="disabled"
          >
                    <a
                              href="#!"
                              onClick={[Function]}
                    >
                              <i
                                        className="material-icons"
                              >
                                        chevron_left
                              </i>
                    </a>
          </li>
          <li
                    className="active"
          >
                    <a
                              href="#!"
                              onClick={[Function]}
                    >
                              1
                    </a>
          </li>
          <li
                    className="disabled"
          >
                    <a
                              href="#!"
                              onClick={[Function]}
                    >
                              <i
                                        className="material-icons"
                              >
                                        chevron_right
                              </i>
                    </a>
          </li>
</ul>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="disabled"
>
              <a
                            href="#!"
                            onClick={[Function]}
              >
                            <i
                                          className="material-icons"
                            >
                                          chevron_left
                            </i>
              </a>
</li>,
            Array [
              <li
                className="active"
>
                <a
                                href="#!"
                                onClick={[Function]}
                >
                                1
                </a>
</li>,
            ],
            <li
              className="disabled"
>
              <a
                            href="#!"
                            onClick={[Function]}
              >
                            <i
                                          className="material-icons"
                            >
                                          chevron_right
                            </i>
              </a>
</li>,
          ],
          "className": "pagination center m-b--20",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#!"
                onClick={[Function]}
>
                <i
                                className="material-icons"
                >
                                chevron_left
                </i>
</a>,
              "className": "disabled",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="material-icons"
>
                  chevron_left
</i>,
                "href": "#!",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "chevron_left",
                  "className": "material-icons",
                },
                "ref": null,
                "rendered": "chevron_left",
                "type": "i",
              },
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#!"
                onClick={[Function]}
>
                1
</a>,
              "className": "active",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "href": "#!",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 1,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="#!"
                onClick={[Function]}
>
                <i
                                className="material-icons"
                >
                                chevron_right
                </i>
</a>,
              "className": "disabled",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="material-icons"
>
                  chevron_right
</i>,
                "href": "#!",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "chevron_right",
                  "className": "material-icons",
                },
                "ref": null,
                "rendered": "chevron_right",
                "type": "i",
              },
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
